app/code/community/Ebizmarts/MailChimp/Block/Checkout/Success/Groups.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+/**
+ * Checkout subscribe checkbox block renderer
+ *
+ * @category   Ebizmarts
+ * @package    Ebizmarts_MageMonkey
+ * @author     Ebizmarts Team <info@ebizmarts.com>
+ * @license    http://opensource.org/licenses/osl-3.0.php
+ */
+class Ebizmarts_MailChimp_Block_Checkout_Success_Groups extends Mage_Core_Block_Template
+{
+
+    protected $_lists = array();
+    protected $_info = array();
+    protected $_myLists = array();
+    protected $_generalList = array();
+    protected $_form;
+    protected $_api;
+    /**
+     * @var Ebizmarts_MailChimp_Helper_Data
+     */
+    protected $helper;
+    protected $storeId;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->helper = Mage::helper('mailchimp');
+        $this->storeId = Mage::app()->getStore()->getId();
+    }
+
+    /**
+     * Get list data from MC
+     *
+     * @return array
+     */
+    public function getGeneralList()
+    {
+        $storeId = $this->storeId;
+        $helper = $this->getMailChimpHelper();
+        $listId = $helper->getGeneralList($storeId);
+
+        return $listId;
+    }
+
+    public function getListInterestGroups()
+    {
+        $storeId = $this->storeId;
+        $helper = $this->getMailChimpHelper();
+        $return = $helper->getListInterestGroups($storeId);
+        return $return;
+    }
+
+    public function getCategoryTitle($category)
+    {
+        return $category['title'];
+    }
+
+    public function renderGroups($category)
+    {
+        $object = $this->createObject($category);
+
+        $this->addGroupOptions($category, $object);
+
+        $html = $this->getElementHtml($category, $object);
+
+        return $html;
+    }
+
+    public function getGroupClass($type)
```

Comment:
this could be improved with a factory method https://designpatternsphp.readthedocs.io/en/latest/Creational/FactoryMethod/README.html